### DISCUSSION TITLE
静态IP无效

### DISCUSSION THREAD

**Discussion by lmj258** [coolsnowwolf/lede#13247] [Q&A / 快速问答]
最新版本的源码编译出来以后，路由器里给设备分配了静态IP，MAC地址也没填错，但是设备接入以后，路由器还是随机分配IP，是哪里的问题，还有编译之前怎样修改lan口默认地址，在package/base-files/files/bin/config_generate文件里修改了，编译出来没有效果

**Comment by 19901019** [comment]
修改lan默认路径package/base-files/luci2/bin/config_generate

**Reply by lmj258** [reply]
谢谢，今晚试试，静态地址分配的问题有解吗？

**Reply by 19901019** [reply]
你把手机IP静态看看，然后手机设置自动获取看看是不是静态，要是手机可以但其它设备不行就要等随机IP十二小时过后静态才生效。

**Reply by lmj258** [reply]
> 你把手机IP静态看看，然后手机设置自动获取看看是不是静态，要是手机可以但其它设备不行就要等随机IP十二小时过后静态才生效。

手机不是，特意用新手机试过，所有设备都不是，等12小时也是随机分配，之前编译的固件就重启路由器就生效，这次的死活不行

**Reply by lmj258** [reply]
> 修改lan默认路径package/base-files/luci2/bin/config_generate

这个貌似不行，改了这个里面的，编译出来network文件直接是空的

**Reply by 19901019** [reply]
> > 修改lan默认路径package/base-files/luci2/bin/config_generate
> 
> 这个貌似不行，改了这个里面的，编译出来network文件直接是空的

你的luci是23.05吗？23.05我是修改这个是可以的

**Reply by lmj258** [reply]
> > > 修改lan默认路径package/base-files/luci2/bin/config_generate
> > 
> > 
> > 这个貌似不行，改了这个里面的，编译出来network文件直接是空的
> 
> 你的luci是23.05吗？23.05我是修改这个是可以的

不知道是啥，不懂，直接拉的最新源码昨晚修改了这个文件里的地址为192.168.2.1编译的，然后今天安装起来，192.168.2.1进不去，运行vi etc/confin/network想自己改以下，打开全是空的
